### PR TITLE
Disallow text transform in stylelint-polaris coverage rules

### LINKED ISSUE #8241: Add text-transform as a failure in polaris-stylelint
## Issue summary

Developers shouldn't be using `text-transform` in the admin. They should be updating translation files or using the Text component without capitalisation. This issue would enforce this direction.

## Expected behavior

Developers get an error when using `text-transform`

## Actual behavior

Developers are using `text-transform` everywhere

### PR DESCRIPTION
**Pull request by sam-b-rose**
### WHY are these changes introduced?

Fixes #8241 

### WHAT is this pull request doing?

Adds the `text-transform` to the disallow properties list of the `stylelint-polaris` plugin.

### CODE REVIEW

**Comment by sam-b-rose** [comment]
@alex-page we can scope the changes to only inserting the disable comments if that is preferred for now. I have two commits, one with the [disable comments](https://github.com/Shopify/polaris/pull/8244/commits/ab5a6c954c58490e7b452648c3b0fa3f62418e62) and the latest with the text-transform styles removed. If we think the disable comments is the way to go, we can revert [this commit](https://github.com/Shopify/polaris/pull/8244/commits/cbe12feb6a07e060533b10d247145bf5f224bde2)

File: polaris-react/src/components/Button/Button.scss [hunk 1]
```diff
@@ -463,6 +462,7 @@
     font-size: var(--p-font-size-200);
     font-weight: var(--p-font-weight-medium);
     line-height: var(--p-font-line-height-2);
+    // stylelint-disable-next-line -- generated by polaris-migrator DO NOT COPY
     text-transform: initial;
```
Comment (alex-page) [inline polaris-react/src/components/Button/Button.scss:468]:
```suggestion
```

Comment (sam-b-rose) [reply polaris-react/src/components/Button/Button.scss:468]: How'd I miss this!! Thanks Alex!

Comment (alex-page) [reply polaris-react/src/components/Button/Button.scss:468]: I dunno lol

**Review by alex-page — APPROVED** [review]

**Pull request merged**